File: API/beONHR.Entities/EmployeeSalary.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace beONHR.Entities
{
    public class EmployeeSalary
    {
        [Key]
        public Guid Id { get; set; }
        public string SalaryType { get; set; }
        public string Amount { get; set; }
        public Guid Currency { get; set; }
        public DateOnly StartDate {  get; set; }
        public DateOnly EndDate {  get; set; }
        public bool IsDeleted { get; set; } = false;
        [ForeignKey("Currency")]
        public virtual Currency? currencyId { get; set; }      

     
    }
} 

File: UI/src/app/shared/models/employee-salary.ts
import { Currency } from "./currency";

export interface EmployeeSalary {
    id: string;
    salaryType: string;
    amount: string;
    currency: string; // Assuming string type for foreign key, change to appropriate type if needed
    startDate: string; // Change to appropriate date type if needed
    endDate: string; // Change to appropriate date type if needed
    currencyDetails?: Currency | null;
  }